bpf/bpf_lxc.c

```diff
@@ -995,6 +1001,16 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx, __u32 *d
 		/* Did we end up at a stale non-service entry? Recreate if so. */
 		if (unlikely(ct_state->rev_nat_index != ct_state_new.rev_nat_index))
 			goto ct_recreate4;
+
+		/* Recreate the CT entry for non-tcp if the proxy_redirect flag is stale.
+		 * Otherwise the return packet will be erroneously redirected (or not)
+		 * For tcp, ct_lookup returns CT_NEW and the caller recreates the entry
+		 * if the packet hits a closing stale entry.
+		 */
+		ct_state_new.proxy_redirect = proxy_port > 0;
+		if (unlikely(tuple->nexthdr != IPPROTO_TCP &&
+			     ct_state->proxy_redirect != ct_state_new.proxy_redirect))
+			goto ct_recreate4;
```

Comment:
I would just drop the check for `IPPROTO_TCP`. There's barely any computational overhead, but adding a conditional has potential to trip the verifier / make verification more expensive.

This would also allows us to handle the case where an **active** TCP connection suddenly comes into scope of a L7 policy. I suppose atm we would just expect the connection to be RST (because we suddenly start to pass its packets through the proxy), and then recover once the SYN is sent?

Reply:
Make sense. I will drop the proto check